### PR TITLE
Rewrite of VoteIT to use Gamma instead of codes.

### PR DESCRIPTION
**Pull request by Portals**
Hello! 

This past sektionsmöte, the IT division used [Sveroks VoteIT](https://sverok.voteit.se/) to conduct an online sektionsmöte. 

This rewrite presents a new solution to conduct online voting via authentication with [Gamma](https://github.com/cthit/gamma) while using Zoom to conduct a sektionsmöte. It will still be completely anonymous. Instead of checking against codes, each vote session will keep track of who has already voted.

To keep track of who are able to vote, every person has to be approve by the counter of the meeting. 

This rewrite also upgrades the way the frontend is compiled and served. Instead of using gulp, create-react-app is used. Less configuration and less pain. I also rewrote the entire frontend to use [react-digit-components](https://github.com/cthit/react-digit-components).

Below are some screenshots of how the new client rewrite looks like:

 
image
image
image
image
image
image
image
image

### CODE REVIEW

**Comment by Portals** [comment]
Before possible merge:in this to `master`. `master` should be merged to `master-physical`.

File: client/src/use-cases/vote/screens/cast-vote/components/candidate-list/CandidateList.component.jsx [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    grid-auto-rows: auto;
+`;
+
+const CandidateList = ({ value, innerInputs }) => {
+    const [status] = useStatus();
+    const candidatesChoosen = useMemo(
+        () =>
+            value.reduce((n, candidate) => (candidate.choosen ? n + 1 : n), 0),
+        /*eslint-disable */
+        [JSON.stringify(value)] //Stringify is neccessary since just using value won't update properly
```
Comment (lindskogen) [inline client/src/use-cases/vote/screens/cast-vote/components/candidate-list/CandidateList.component.jsx:19]: Why?

File: client/src/use-cases/vote/screens/cast-vote/components/candidate-list/CandidateList.component.jsx [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    const lowestVacantIndex = useMemo(() => {
+        const vacantCandidate = find(
+            value.filter(candidate => candidate.vacant),
+            ["choosen", false]
+        );
+
+        return vacantCandidate == null ? -1 : vacantCandidate.id;
+        /*eslint-disable */
+    }, [JSON.stringify(value)]);
```
Comment (lindskogen) [inline client/src/use-cases/vote/screens/cast-vote/components/candidate-list/CandidateList.component.jsx:31]: Why?

File: client/public/robots.txt [hunk 3]
```diff
@@ -0,0 +1,3 @@
+# https://www.robotstxt.org/robotstxt.html
+User-agent: *
+Disallow:
```
Comment (williamleven) [inline client/public/robots.txt]: Do you intend for robots to visit the site?, if not `Disallow: /` is the way to go

File: index.js [hunk 4]
```diff
@@ -0,0 +1,354 @@
...
+    if (production && req.headers["x-forwarded-proto"] !== "https") {
+        return res.end("Please visit the site with HTTPS");
+    } else {
+        next();
+    }
+});
+
+app.use(express.urlencoded({ extended: true }));
+app.use(express.json());
+app.use(cors());
```
Comment (williamleven) [inline index.js:88]: You probably don't want to allow cors in production

File: client/public/robots.txt (hunk 3 above)
Comment (Portals) [reply client/public/robots.txt]: No, missed checking this when `create-react-app` generated its code.

File: index.js [hunk 5]
```diff
@@ -0,0 +1,363 @@
...
+                path: "/",
+                expires: new Date(Date.now() + 24 * 3600000), //24h
+                httpOnly: true,
+                secure: production,
+                sameSite: "strict",
+            });
+
+            const userId = user.id;
+
+            if (req.user.cid === app.get("adminCid")) {
```
Comment (Deltaphish) [inline index.js:199]: Keep getting `TypeError: Cannot read property 'cid' of undefined
    at /VoteIT/index.js:199:26` when logging in.

Still works, but checking if cid is defined would be nice.

**Pull request closed without merging**